### ISSUE TITLE
Immutable GAPInfo.PackagesInfo records?

### ISSUE THREAD

**Issue by ThomasBreuer** [gap-system/gap#2571]
Since #2510, the records stored in `GAPInfo.PackagesInfo` are immutable.
Until then, I was able to modify such a record in place from the package in question.
This was useful in the following situation, in the package `JuliaInterface`:
The package loads a kernel module, and afterwards GAP has access to the Julia system.
It is useful to know and show the version of Julia one is working with,
and a natural place for that is the banner string that is shown by the package `JuliaInterface`.
The natural way to determine the version number is to ask Julia itself for it,
and then the version number can be inserted into this string only after the kernel module has been loaded.
Thus my idea was to modify the banner string in the package's `read.g` file.
This worked well until #2510.

Now the banner string is immutable.
Assigning a modified version to the appropriate component of `GAPInfo.PackagesInfo` (a dirty hack)
would not help because the relevant function (`LoadPackage_ReadImplementationParts`) uses its own list of records.

What do you propose?

**Comment by fingolfin** [comment]
This is a duplicate of issue #2568

**Comment by stevelinton** [comment]
Since discussion in #2568 has already progressed further (at a glance I like @laurentbartholdi's proposal. I've renamed that issue to make the situation clearer and I'm closing this one.
